src/main/java/com/teragrep/pth_06/planner/offset/SerializedDatasourceOffset.java

```diff
@@ -61,27 +61,55 @@ public class SerializedDatasourceOffset implements Serializable {
 
     private final Long version = 1L;
 
+    public final HdfsOffset hdfsOffset;
     public final LongOffset archiveOffset;
     public final KafkaOffset kafkaOffset;
 
+    public SerializedDatasourceOffset(HdfsOffset hdfsOffset, LongOffset archiveOffset, KafkaOffset kafkaOffset) {
+        this.hdfsOffset = hdfsOffset;
+        this.archiveOffset = archiveOffset;
+        this.kafkaOffset = kafkaOffset;
+    }
+
     public SerializedDatasourceOffset(LongOffset archiveOffset, KafkaOffset kafkaOffset) {
+        this.hdfsOffset = null;
+        this.archiveOffset = archiveOffset;
+        this.kafkaOffset = kafkaOffset;
```

Comment:
same refactoring comment with this, but perhaps for another issue?

Reply:
Solved in commit 275a5bbc898dabbd855844583eaff82c91b2c3da